src/services/EthService/utils/transactions.ts

```diff
@@ -50,10 +50,16 @@ export const appendGasLimit = (network: Network) => async (
   tx: TxBeforeGasLimit
 ): Promise<TxBeforeNonce> => {
   try {
-    const gasLimit = await getGasEstimate(network, tx)
-      .then(hexToNumber)
-      .then((n: number) => Math.round(n * 1.2))
-      .then((n: number) => inputGasLimitToHex(n.toString()));
+    const gasLimit = await (async () => {
+      // Respect gas limit if present
+      if (tx.gasLimit) {
+        return tx.gasLimit;
+      }
+      return await getGasEstimate(network, tx)
+        .then(hexToNumber)
+        .then((n: number) => Math.round(n * 1.2))
+        .then((n: number) => inputGasLimitToHex(n.toString()));
+    })();
```

Comment:
We should be able to dispense with IIFE here.
A simple guard should do the trick
```
if (tx.gasLimit) return {...tx}
try {
  ...
} catch {


}
```